api/scaffold/__init__.py

```diff
@@ -1781,25 +1847,26 @@ def __init__(
 
         # Hardware version module. Defined here because it is an architecture
         # requirement. There is no need to expose this module.
-        self.__version_module = Version(self)
+        if isinstance(self, Scaffold):
+            self.__version_module = Version(self)
```

Comment:
Why adding a test here?
`__version_module` shall always be present.

Reply:
Because of line 223:
def __init__(self, parent: "Scaffold"):

I would suggest then to change parent type to "ArchBase".

```suggestion
        self.__version_module = Version(self)

...

class Version(Module):
    """Version module of the board."""

    def __init__(self, parent: "ArchBase"):
```

Reply:
Also it should be propagated to other classes (Module, Signal, ...).

Reply:
Yes `ArchBase` is the correct type.
About propagation maybe we can report that for later, I'm not sure if this is good yet.

Reply:
So, I suggest to let "Scaffold" and add a "type: ignore" comment.
```suggestion
        # Hardware version module. Defined here because it is an architecture
        # requirement. There is no need to expose this module.
        self.__version_module = Version(self)  # type: ignore

```